File: Oldbook/oldbook.py
import sys
from math import gcd

fi = open("input.txt")
fo = open("output.txt", "w")
sys.stdin = fi
sys.stdout = fo


def xuat_kq():
    global ans
    import os
    print(ans)
    fo.close()
    fi.close()
    os._exit(os.F_OK)


n = int(input())
s = input()
k = int(input())
b, mn, mx = [False]*26, 'z', 'a'

for c in s:
    b[ord(c) - 97] = True
    if mx < c:
        mx = c
    if mn > c:
        mn = c

if n < k:
    ans = s
    for i in range(n - k):
        ans += mn
    xuat_kq()

p = k - 1
while p >= 0 and s[p] == mx:
    p -= 1

for i in range(ord(s[p]) - 97 + 1, 26):
    if b[i]:
        q = chr(i + 97)
        break

ans = s[:p] + q
for i in range(p + 1, k):
    ans += mn

xuat_kq()

File: Oldbook/oldbook.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, k;
    string s, ans;
    char mn = 'z', mx = 'a';
    cin >> n >> s >> k;
    bool b[26] = {0};
    for (char c : s)
    {
        if (c > mx)
            mx = c;
        if (c < mn)
            mn = c;
        b[c - 97] = true;
    }

    if (n < k)
    {
        ans = s;
        for (int i = 0; i < k - n; i++)
        {
            ans += mn;
        }
        cout << ans << endl;
        return 0;
    }

    int p = k - 1, q;
    while (p >= 0 && s[p] == mx)
    {
        --p;
    }
    for (int i = s[p] - 97 + 1; i < 26; i++)
    {
        if (b[i])
        {
            q = i;
            break;
        }
    }
    ans = s.substr(0, p) + char(q + 97);
    for (int i = p + 1; i < k; i++)
    {
        ans += mn;
    }
    cout << ans << endl;

    return 0;
}